src/software/new_geom/util/intersection.cpp

```diff
@@ -54,11 +54,11 @@ std::vector<Point> intersection(const Segment &first, const Segment &second)
     return output;
 }
 
-std::unordered_set<Point> intersection(const Rectangle &rectangle, const Segment &segment)
+std::unordered_set<Point> intersection(const Polygon &polygon, const Segment &segment)
 {
     std::unordered_set<Point> intersections;
 
-    for (const Segment &rec_segment : rectangle.getSegments())
+    for (const Segment &rec_segment : polygon.getSegments())
```

Comment:
`rec_segment` -> `segment` (or `seg` to be consistent with what we're doing below)

Reply:
Fixed in latest